src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java

```diff
@@ -1864,6 +1865,12 @@ public boolean canSubmitImprovementMarkSheets(final ExecutionYear executionYear)
         return sortedExecutionDegrees.last().getExecutionYear().equals(executionYear.getPreviousExecutionYear());
     }
 
+    public ExecutionInterval getBegin() {
+        final List<ExecutionYear> executionYears = new ArrayList<ExecutionYear>(getBeginContextExecutionYears());
```

Comment:
This seems like a really wasteful way of doing this. You should use [Collections.min](http://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#min-java.util.Collection-java.util.Comparator-) or something similar instead.